packages/yarnpkg-core/tests/Configuration.test.ts

```diff
@@ -1,8 +1,35 @@
-import {xfs, PortablePath}                 from '@yarnpkg/fslib';
-import NpmPlugin                           from '@yarnpkg/plugin-npm';
-
-import {Configuration, SECRET, TAG_REGEXP} from '../sources/Configuration';
-
+import {structUtils}                                                         from '@yarnpkg/core';
+import {npath, xfs, PortablePath}                                            from '@yarnpkg/fslib';
+import {stringifySyml}                                                       from '@yarnpkg/parsers';
+import NpmPlugin                                                             from '@yarnpkg/plugin-npm';
+
+import {Configuration, SECRET, SettingsDefinition, SettingsType, TAG_REGEXP} from '../sources/Configuration';
+
+const initConfigurationPlugin = async (configuration: {[coreSettingName: string]: SettingsDefinition}) => {
+  const CONFIGURATION_PLUGIN = `
+    const factory = r => {
+      return {
+        default: {
+          configuration: ${JSON.stringify(configuration)},
```

Comment:
May be more performant here to have it do `JSON.parse` at runtime instead of using an object literal - especially for big configs.

Reply:
Not sure if I got it right 🥸

I changed the type of configuration to a string

https://github.com/yarnpkg/berry/pull/4982/commits/0a4f244aeb188740dbabc29ed4e3760896c515ec#diff-7841ca345244c29f52d5e666b712534e41ec3ae18674a61d093180edb1a8926eR8